fe/fe-core/src/main/java/org/apache/doris/datasource/property/metastore/HMSBaseProperties.java

```diff
@@ -198,8 +199,11 @@ private void checkAndInit() {
         if (StringUtils.isNotBlank(hmsUserName)) {
             hiveConf.set(AuthenticationConfig.HADOOP_USER_NAME, hmsUserName);
         }
-        HiveConf.setVar(hiveConf, HiveConf.ConfVars.METASTORE_CLIENT_SOCKET_TIMEOUT,
-                String.valueOf(Config.hive_metastore_client_timeout_second));
+        if (!userOverriddenHiveConfig.containsKey(ConfVars.METASTORE_CLIENT_SOCKET_TIMEOUT.toString())) {
```

Comment:
`userOverriddenHiveConfig` will overwrite this propertiy. 
Just put this before `initUserHiveConfig` and it will take effect.